GitHub Actions are failing on Windows on Node.js v26

Fixed by:
* #3331

https://github.com/nodejs/node-gyp/actions/workflows/tests.yml

Same here:

https://github.com/vrza/node-blake2/actions/runs/27128279103/job/80210657406

Build fails due to invalid flags in the generated .vcxproj -- on first glance looks like LLVM tools' flags are passed to MSVC tools.

```
LINK : warning LNK4044: unrecognized option '/flto=thin'; ignored [D:\a\node-blake2\node-blake2\build\blake2.vcxproj]
LINK : fatal error LNK1117: syntax error in option 'opt:lldltojobs=2' [D:\a\node-blake2\node-blake2\build\blake2.vcxproj]
...

--- a/lib/create-config-gypi.js
+++ b/lib/create-config-gypi.js
@@ -99,6 +99,11 @@ async function getCurrentConfigGypi ({ gyp, nodeDir, vsInfo, python }) {
     if (config.variables.clang === 1) {
       config.variables.clang = 0
     }
+    // disable LTO for addon builds. node release builds may enable (thin) LTO,
+    // which leaks clang/lld-only flags like -flto=thin and /opt:lldltojobs into
+    // addons built with the default MSVC toolchain, which rejects those flags
+    variables.enable_lto = 'false'
+    variables.enable_thin_lto = 'false'
   }
 
   // loop through the rest of the opts and add the unknown ones as variables.

--- a/test/fixtures/win-lto/include/node/config.gypi
+++ b/test/fixtures/win-lto/include/node/config.gypi
@@ -1,0 +1,8 @@
+# Test configuration simulating a Node.js Windows release built with Thin LTO
+{
+  'variables': {
+    'enable_lto': 'true',
+    'enable_thin_lto': 'true',
+    'lto_jobs': '2'
+  }
+}

--- a/test/test-create-config-gypi.js
+++ b/test/test-create-config-gypi.js
@@ -52,6 +52,25 @@ describe('create-config-gypi', function () {
     assert.strictEqual(config.variables.build_with_electron, undefined)
   })
 
+  it('config.gypi disables LTO for addon builds on Windows', async function () {
+    const nodeDir = path.join(__dirname, 'fixtures', 'win-lto')
+
+    const prog = gyp()
+    prog.parseArgv(['_', '_', `--nodedir=${nodeDir}`])
+
+    const originalPlatform = process.platform
+    Object.defineProperty(process, 'platform', { value: 'win32' })
+    try {
+      const config = await getCurrentConfigGypi({ gyp: prog, nodeDir, vsInfo: {} })
+      // thin LTO leaks clang/lld-only flags into the MSVC addon build, so it
+      // must be disabled regardless of how node itself was built
+      assert.strictEqual(config.variables.enable_lto, 'false')
+      assert.strictEqual(config.variables.enable_thin_lto, 'false')
+    } finally {
+      Object.defineProperty(process, 'platform', { value: originalPlatform })
+    }
+  })
+
   it('config.gypi parsing', function () {
     const str = "# Some comments\n{'variables': {'multiline': 'A'\n'B'}}"
     const config = parseConfigGypi(str)

